packages/qgrid-ngx-plugins/src/lib/data-provider/strategies/cache-requested-pages.ts

```diff
@@ -0,0 +1,44 @@
+import { Observable, of } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+import { DataProviderStrategy } from '../data-provider-strategy';
+import { DataProviderServer } from '../models';
+
+export class CacheAlreadyRequestedPageStrategy<T> extends DataProviderStrategy<T> {
+	private gridRowsCache: Map<number, T[]> = new Map();
+
+	constructor(
+		private server: Pick<DataProviderServer<T>, 'getPage'>,
+		private pagesToLoad: number = 0,
+	) {
+		super();
+	}
+
+	processData(memo: T[]): Observable<T[]> {
+		const { current, size } = this.gridModel.pagination();
+
+		if (this.pagesToLoad) {
+			this.loadInBackground(this.pagesToLoad);
+		}
+
+		if (this.gridRowsCache.has(current)) {
+			return of(this.gridRowsCache.get(current));
+		}
+
+		const shouldRequestData = !memo?.length;
+		return (shouldRequestData ? this.server.getPage(current, size) : of(memo))
+			.pipe(tap(rows => this.gridRowsCache.set(current, rows)));
+	}
+
+	private loadInBackground(pagesToLoad: number): void {
+		const { current, size } = this.gridModel.pagination();
```

Comment:
why we just wont' do a one server call to loadd required number of pages? maybe it should be an option?